django/core/files/storage.py

```diff
@@ -324,6 +339,25 @@ def created_time(self, name):
     def modified_time(self, name):
         return datetime.fromtimestamp(os.path.getmtime(self.path(name)))
 
+    def _do_safe_operation(self, operation, src_path, dst_path):
+        try:
+            operation(src_path, dst_path)
+        except OSError as e:
+            if e.errno == errno.EEXIST:
+                raise ValueError('Destination already exists')
```

Comment:
The way I see it there are multiple alternatives:
* use this generic `ValueError` exception which has the meaning that your argument is invalid for the reason highlighted in the error message
* create a dedicated exception in the storage module and use that. Something along `django.core.files.storage.FileExistsError` and use that
* simply not specify what the behaviour should be and leave that up to the various implementations. This has the downside of leaving the API's clients having to handle different failure modes depending on which particular storage backend they use and therefore result in leaky abstractions.

If I think about it, the last option above would probably make the most sense. This is because you might have certain storage backends that can't atomically decide whether a file exists and also perform the move/copy. Having to do two separate requests would result in concurrency issues. These kind of backends would simply not be able to safely implement the method as requested by the API.